src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -684,9 +684,9 @@ internal static char ParseFormatSpecifier(ReadOnlySpan<char> format, out int dig
                         // digits.  Further, for compat, we need to stop when we hit a null char.
                         int n = 0;
                         int i = 1;
-                        while (i < format.Length && (((uint)format[i] - '0') < 10) && n < 10)
+                        while (i < format.Length && (((uint)format[i] - '0') < 10))
                         {
-                            n = (n * 10) + format[i++] - '0';
+                            n = checked((n * 10) + format[i++] - '0');
```

Comment:
We could now be throwing an OverflowException where we weren't before?  Should this be translated somewhere into a FormatException?

Reply:
I thought about it and decided to leave it as an OverflowException. Do you think it affects usability? The plan is to update the documentation here to state that a precision value > int.Max will throw an OverflowException.

Reply:
I think it might be better to keep it as `FormatException` for simplicity, that way users don't need to update even more paths.

We can always do something like:
```csharp
var tmp = (n * 10)+ format[i++] - '0'

if (tmp < n)
{
    ThrowFormatException();
}

n = tmp;
```